### PR TITLE
[OpenXLA] Add openxla.md

### PR DESCRIPTION
**Pull request by ManfeiBai**

### CODE REVIEW

File: docs/openxla.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# OpenXLA
+
+PyTorch/XLA is a Python package that uses the [XLA deep learning
+compiler](https://www.tensorflow.org/xla) to connect the [PyTorch deep learning
+framework](https://pytorch.org/) and [Cloud
+TPUs](https://cloud.google.com/tpu/). PyTorch/XLA now pull XLA from OpenXLA. 
+OpenXLA here means [open source machine learning compiler XLA for GPUs, CPUs, and ML accelerators](https://github.com/openxla/xla), 
+which belongs to the same-name community [OpenXLA Community](https://github.com/openxla) and this community also contain [StableHLO](https://github.com/openxla/stablehlo).
```
Comment (wonjoolee95) [inline docs/openxla.md:8]: Let's rephrase this as:

```
As of ***<insert_date_where_we_pushed_openxla_migration>***, PyTorch/XLA now ***pulls*** XLA from OpenXLA. 
OpenXLA ***is an*** [open source machine learning compiler XLA for GPUs, CPUs, and ML accelerators](https://github.com/openxla/xla). 
```

Let's remove the sentence `which belongs to the same-name community [OpenXLA Community](https://github.com/openxla) and this community also contain [StableHLO](https://github.com/openxla/stablehlo).`  since this seems redundant.

File: docs/openxla.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+For [PJRT runtime](https://github.com/pytorch/xla/blob/master/docs/pjrt.md) user,
+there is no logic change before and after [Migrate PyTorch/XLA to pull XLA from OpenXLA](https://github.com/pytorch/xla/pull/5202).
+For XRT runtime user, because OpenXLA don't support XRT, user could use XRT from [XRT branch of PyTorch/XLA](https://github.com/pytorch/xla/tree/xrt).
+
+# Difference
+
+There is no change in user side. Before migrate to pull XLA from OpenXLA, PyTorch/XLA need to build with TensorFlow, but now build with OpenXLA
+
+# Performance
+| Training Throughput on ResNet50 |
```
Comment (wonjoolee95) [inline docs/openxla.md:26]: Is this formatted correctly? The title "Training Throughput on ResNet50" doesn't seem to be inside the table.

File: docs/openxla.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+|    | resnet50-pjrt-v2-8 | resnet50-pjrt-v4-8 | resnet50-pjrt-v4-32 |
+| -------- | ------- | ------- | ------- |
+| Pre Migration  | 18.59    | 20.06 | 27.92 |
+| Post Migration | 18.63    | 19.94 | 27.14 |
+
+# pin dependency sunset
+PyTorch/XLA pull XLA from OpenXLA via pin to a specific commit
+
+# other `OpenXLA` or `openxla` used in PyTorch or PyTorch/XLA
```
Comment (wonjoolee95) [inline docs/openxla.md:35]: What is this for?

File: docs/openxla.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+There is no change in user side. Before migrate to pull XLA from OpenXLA, PyTorch/XLA need to build with TensorFlow, but now build with OpenXLA
+
+# Performance
+| Training Throughput on ResNet50 |
+|    | resnet50-pjrt-v2-8 | resnet50-pjrt-v4-8 | resnet50-pjrt-v4-32 |
+| -------- | ------- | ------- | ------- |
+| Pre Migration  | 18.59    | 20.06 | 27.92 |
+| Post Migration | 18.63    | 19.94 | 27.14 |
+
+# pin dependency sunset
```
Comment (wonjoolee95) [inline docs/openxla.md:32]: This section doesn't seem to be complete.

**Review by wonjoolee95 — CHANGES_REQUESTED** [review]

File: docs/openxla.md (hunk 3 above)
Comment (ManfeiBai) [reply docs/openxla.md:35]: I remember we use term `openxla` as the backend in Dynamo, so wanna mention it here

File: docs/openxla.md (hunk 4 above)
Comment (ManfeiBai) [reply docs/openxla.md:33]: thanks, updated the table, will update table again in the future after fully compared performance

File: docs/openxla.md (hunk 2 above)
Comment (ManfeiBai) [reply docs/openxla.md]: thanks, update the table

File: docs/openxla.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+# How to use OpenXLA
+
+For [PJRT runtime](https://github.com/pytorch/xla/blob/master/docs/pjrt.md) user,
+there is no logic change before and after [Migrate PyTorch/XLA to pull XLA from OpenXLA](https://github.com/pytorch/xla/pull/5202).
+For XRT runtime user, because OpenXLA don't support XRT, user could use XRT from [XRT branch of PyTorch/XLA](https://github.com/pytorch/xla/tree/xrt).
+
+# Difference
+
+There is no change in user side. Before migrate to pull XLA from OpenXLA, PyTorch/XLA need to build with TensorFlow, but now build with OpenXLA
```
[the remark this answers is not in the document]
Comment (ManfeiBai) [reply docs/openxla.md]: thanks, updated

File: docs/openxla.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+PyTorch/XLA used to pull XLA from [TensorFlow](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/compiler/xla) before https://github.com/pytorch/xla/pull/5202.
+After https://github.com/pytorch/xla/pull/5202, PyTorch/XLA has migrated from pull XLA from [TensorFlow](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/compiler/xla)
+to pull XLA from [OpenXLA](https://github.com/openxla/xla).
+
+
+# How to use OpenXLA
+
+For [PJRT runtime](https://github.com/pytorch/xla/blob/master/docs/pjrt.md) user,
+there is no logic change before and after [Migrate PyTorch/XLA to pull XLA from OpenXLA](https://github.com/pytorch/xla/pull/5202).
+For XRT runtime user, because OpenXLA don't support XRT, user could use XRT from [XRT branch of PyTorch/XLA](https://github.com/pytorch/xla/tree/xrt).
```
[the remark this answers is not in the document]
Comment (ManfeiBai) [reply docs/openxla.md]: thanks, updated

File: docs/openxla.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+PyTorch/XLA is a Python package that uses the [XLA deep learning
+compiler](https://www.tensorflow.org/xla) to connect the [PyTorch deep learning
+framework](https://pytorch.org/) and [Cloud
+TPUs](https://cloud.google.com/tpu/). PyTorch/XLA now pull XLA from OpenXLA. 
+OpenXLA here means [open source machine learning compiler XLA for GPUs, CPUs, and ML accelerators](https://github.com/openxla/xla), 
+which belongs to the same-name community [OpenXLA Community](https://github.com/openxla) and this community also contain [StableHLO](https://github.com/openxla/stablehlo).
+
+PyTorch/XLA used to pull XLA from [TensorFlow](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/compiler/xla) before https://github.com/pytorch/xla/pull/5202.
+After https://github.com/pytorch/xla/pull/5202, PyTorch/XLA has migrated from pull XLA from [TensorFlow](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/compiler/xla)
+to pull XLA from [OpenXLA](https://github.com/openxla/xla).
```
[the remark this answers is not in the document]
Comment (ManfeiBai) [reply docs/openxla.md:12]: thanks, updated

File: docs/openxla.md (hunk 1 above)
Comment (ManfeiBai) [reply docs/openxla.md]: thanks, updated

File: docs/openxla.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
+# OpenXLA
+
+As of ***Jun 28***, PyTorch/XLA now ***pulls*** XLA from OpenXLA.
```
Comment (wonjoolee95) [inline docs/openxla.md:3]: Oh, I meant to highlight certain words that I wanted to mention in the comment with the `***`. You can remove all the `***`s here. 

nit: Let's add the year, too -- June 28th, 2023

File: docs/openxla.md [hunk 9]
```diff
@@ -0,0 +1,25 @@
+# OpenXLA
+
+As of ***Jun 28***, PyTorch/XLA now ***pulls*** XLA from OpenXLA. 
+OpenXLA ***is an*** [open source machine learning compiler XLA for GPUs, CPUs, and ML accelerators](https://github.com/openxla/xla).
```
Comment (wonjoolee95) [inline docs/openxla.md:4]: same as above, remove the `***`

File: docs/openxla.md [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+
+Previous to OpenXLA, PyTorch/XLA pulled XLA directly from [TensorFlow](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/compiler/xla). With our [XLA to OpenXLA migration](https://github.com/pytorch/xla/pull/5202), PyTorch/XLA now pulls XLA from [OpenXLA](https://github.com/openxla/xla).
+
+# How to use OpenXLA
+
+For [PJRT runtime](https://github.com/pytorch/xla/blob/master/docs/pjrt.md) users, there is no change with this migration. For XRT runtime users, there is a separate [XRT branch of PyTorch/XLA](https://github.com/pytorch/xla/tree/xrt) since OpenXLA doesn't support XRT, .
+
+
+# Performance
+||Training Throughput on ResNet50|Training Throughput on ResNet50 |Training Throughput on ResNet50 |
```
Comment (wonjoolee95) [inline docs/openxla.md:14]: Let's actually remove the first row `Training Throughput on ResNet50`. And let's add a small sentence before this, saying:
```
Below is a performance comparison for ResNet50 pre and post the migration on different TPU hardwares. 
```

File: docs/openxla.md (hunk 4 above)
Comment (wonjoolee95) [reply docs/openxla.md:21]: I was referring to this section `pin dependency sunset`. The sentence here doesn't seem complete? We can also just remove this section for now.

File: docs/openxla.md (hunk 3 above)
Comment (wonjoolee95) [reply docs/openxla.md:24]: Let's remove this section.

**Review by wonjoolee95 — CHANGES_REQUESTED** [review]
I should've left these comments in my first review 😢 Thanks for taking care of this.

File: docs/openxla.md (hunk 3 above)
Comment (ManfeiBai) [reply docs/openxla.md]: thanks, updated

File: docs/openxla.md (hunk 4 above)
Comment (ManfeiBai) [reply docs/openxla.md:21]: do we want to create a new doc for `# pin dependency sunset` or we add doc for `# pin dependency sunset` after cut branch?

File: docs/openxla.md (hunk 8 above)
Comment (ManfeiBai) [reply docs/openxla.md]: thanks, updated

File: docs/openxla.md (hunk 9 above)
Comment (ManfeiBai) [reply docs/openxla.md]: thanks, updated

File: docs/openxla.md (hunk 10 above)
Comment (ManfeiBai) [reply docs/openxla.md]: thanks, updated

File: docs/openxla.md (hunk 4 above)
Comment (ManfeiBai) [reply docs/openxla.md]: thanks, updated

File: docs/openxla.md [hunk 11]
```diff
@@ -0,0 +1,19 @@
+# OpenXLA
+
+As of June 28th, 2023, PyTorch/XLA now pulls XLA from OpenXLA. 
+OpenXLA is an [open source machine learning compiler XLA for GPUs, CPUs, and ML accelerators](https://github.com/openxla/xla). 
+
+Previous to OpenXLA, PyTorch/XLA pulled XLA directly from [TensorFlow](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/compiler/xla). With our [XLA to OpenXLA migration](https://github.com/pytorch/xla/pull/5202), PyTorch/XLA now pulls XLA from [OpenXLA](https://github.com/openxla/xla).
+
+# How to use OpenXLA
+
+For [PJRT runtime](https://github.com/pytorch/xla/blob/master/docs/pjrt.md) users, there is no change with this migration. For XRT runtime users, there is a separate [XRT branch of PyTorch/XLA](https://github.com/pytorch/xla/tree/xrt) since OpenXLA doesn't support XRT, .
```
Comment (wonjoolee95) [inline docs/openxla.md:10]: nit: small puncutation typo here at the end of the sentence `, .` -> `.`

Comment (ManfeiBai) [reply docs/openxla.md]: thanks, udpated

**Review by wonjoolee95 — APPROVED** [review]
